virtual-hub/src/graphs.c

```diff
@@ -0,0 +1,90 @@
+#include <stdio.h>
```

Comment:
Could you add license / copyright information in relavant files, at least .c and .h files need this info.